rules/no-hex-escape.js

```diff
@@ -2,7 +2,7 @@
 'use strict';
 
 function checkEscape(context, node, value) {
-	const fixedValue = typeof value === 'string' ? value.replace(/\\x/g, '\\u00') : value;
+	const fixedValue = typeof value === 'string' ? value.replace(/((^|[^\\])(\\\\)*)\\x/g, '$1\\u00') : value;
```

Comment:
@sindresorhus If we use non-capturing groups (`?:`), does it impact performance?